Content.Server/Radiation/Systems/RadiationSystem.cs

```diff
@@ -32,8 +37,101 @@ public override void Initialize()
+        SubscribeLocalEvent<RadiationSourceComponent, ComponentShutdown>(OnSourceShutdown);
+        SubscribeLocalEvent<RadiationSourceComponent, MoveEvent>(OnSourceMove);
+        SubscribeLocalEvent<RadiationSourceComponent, StackCountChangedEvent>(OnSourceStackChanged);
+
+        SubscribeLocalEvent<RadiationReceiverComponent, ComponentInit>(OnReceiverInit);
+        SubscribeLocalEvent<RadiationReceiverComponent, ComponentShutdown>(OnReceiverShutdown);
+    }
+
+    private void OnSourceInit(Entity<RadiationSourceComponent> entity, ref ComponentInit args)
+    {
+        UpdateSource(entity);
+    }
+
+    private void OnSourceShutdown(EntityUid uid, RadiationSourceComponent component, ComponentShutdown args)
+    {
+        if (component.Proxy != DynamicTree.Proxy.Free)
+        {
+            _sourceTree.DestroyProxy(component.Proxy);
+            component.Proxy = DynamicTree.Proxy.Free;
+        }
+        _sourceDataMap.Remove(uid);
+    }
+
+    private void OnSourceMove(Entity<RadiationSourceComponent> entity, ref MoveEvent args)
+    {
+        if (args.NewPosition.EntityId == args.OldPosition.EntityId &&
+            args.NewPosition.Position.EqualsApprox(args.OldPosition.Position))
+            return;
+
+        UpdateSource(entity);
+    }
+
+    private void OnSourceStackChanged(Entity<RadiationSourceComponent> entity, ref StackCountChangedEvent args)
+    {
+        UpdateSource(entity);
+    }
+
+    private void OnReceiverInit(EntityUid uid, RadiationReceiverComponent component, ComponentInit args)
+    {
+        _activeReceivers.Add(uid);
+    }
+
+    private void OnReceiverShutdown(EntityUid uid, RadiationReceiverComponent component, ComponentShutdown args)
+    {
+        _activeReceivers.Remove(uid);
+    }
+
+    private void UpdateSource(Entity<RadiationSourceComponent> entity)
+    {
+        var (uid, component) = entity;
+        var xform = Transform(uid);
+
+        if (!component.Enabled || Terminating(uid))
+        {
+            if (component.Proxy != DynamicTree.Proxy.Free)
+            {
+                _sourceTree.DestroyProxy(component.Proxy);
+                component.Proxy = DynamicTree.Proxy.Free;
+            }
+            _sourceDataMap.Remove(uid);
+            return;
+        }
+
+        var worldPos = _transform.GetWorldPosition(xform);
+        var intensity = component.Intensity * _stack.GetCount(uid);
+        intensity = GetAdjustedRadiationIntensity(uid, intensity);
+
+        if (intensity <= 0)
+        {
+            if (component.Proxy != DynamicTree.Proxy.Free)
+            {
+                _sourceTree.DestroyProxy(component.Proxy);
+                component.Proxy = DynamicTree.Proxy.Free;
+            }
+            _sourceDataMap.Remove(uid);
+            return;
+        }
+
+        var maxRange = component.Slope > 1e-6f ? intensity / component.Slope : GridcastMaxDistance;
+        maxRange = Math.Min(maxRange, GridcastMaxDistance);
```

Comment:
This seems a bit like a magic number, what's stopping this from just being?
```suggestion
        var maxRange = Math.Min(intensity / component.Slope, GridcastMaxDistance);
```

Reply:
This is necessary to avoid division by 0. Because the variable type is float, I did it like this